src/frontend/apps/desk/src/features/members/components/SearchMembers.tsx

```diff
@@ -9,14 +9,30 @@ import { isValidEmail } from '@/utils';
 
 import { KEY_LIST_USER, useUsers } from '../api/useUsers';
 
-export type OptionSelect = Options<{
-  value: Partial<User> & { email: User['email'] };
+export const isOptionNewMember = (
+  data: OptionSelect,
+): data is OptionNewMember => {
+  return 'id' in data.value;
+};
+
+export interface OptionInvitation {
+  value: { email: string };
   label: string;
-}>;
+  type: 'invitation';
+}
+
+export interface OptionNewMember {
+  value: User;
+  label: string;
+  type: 'new_member';
```

Comment:
Could the type be determined dynamically on values' attribute type? Less explicit tho

Reply:
let's keep it as it is, your initial code is much more explicit